### PR TITLE
Infrastructure and an experimental A4A creative.

### PR DESCRIPTION
**Pull request by aghassemi**

### CODE REVIEW

File: src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+}--->
+<!-- -->
+<!doctype html>
+<html ⚡4ads>
+
+<head>
+  <meta charset="utf-8">
+  <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
+
+  <!-- TODO(aghassemi, #9311): Put real binaries when #9311 is resolved -->
```
Comment (sebastianbenz) [inline src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html]: Can you remove this comment as it will show up in the docs. ;-) Or rephrase it to something like:

```
  <!-- ## Setup -->
  <!-- 
  The features used in this sample are still experimental and require a custom version of the AMP runtime.
  -->
```

File: src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+
+  <style amp4ads-boilerplate>
+    body {
+      visibility: hidden
+    }
+  </style>
+
+  <!-- ## Styling -->
+  <!--
+  This is a relatively advanced example that requires some styling to make it
```
Comment (sebastianbenz) [inline src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html]: remove `relatively`

File: src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+  </style>
+</head>
+
+<body>
+  <!-- ## Scrollbound Animations -->
+  <!--
+    There are two types of scrollbound animations that occur in this sample:
+    **scaling** of the video container and **parallax** movement of the background image.
+
+    [amp-animation](https://www.ampproject.org/docs/reference/components/amp-animation)
```
Comment (sebastianbenz) [inline src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html]: Add "The animations are implemented using [amp-animation]..."

File: src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+    is an experimental AMP component that uses [Web Animations API](https://www.w3.org/TR/web-animations/)
+    to support both __time__ and __scroll__ based animations.
```
Comment (sebastianbenz) [inline src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html]: Suggestion: show the amp-animation code first (here) and then explain what's going on. At the moment there is too much text before you can see the code.

File: src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+
+    **scroll vs. time ticker**: Setting `"ticker": "scroll"` tells
+    `amp-animation` to advance the animation timeline based on scrolling rather than time.
+
+    Normally animations have a duration defined, scrollbound animations also have
+    a duration but in pixels rather than milliseconds. Currently duration of
+    a scrollbound animation is approximately one viewport height, meaning all keyframes
+    of the animation will be hit between the time ad is fully in viewport until
+    user scrolls it partially out of the viewport.
```
Comment (sebastianbenz) [inline src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html]: This section is super helpful (and should go into the future amp-animation sample), but might be too much here as it's not really relevant. Wdyt?

File: src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+
+    **Scale down video to the corner when scrolling**:
+    As you have noticed, the video covers the whole ad initially and
+    after the page is scrolled, it slowly starts to scale down and move to the
+    corner reaching its target size when ad is in the middle of the view port.
```
Comment (sebastianbenz) [inline src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html]: nit: s/ad/the ad

File: src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html [hunk 7]
```diff
@@ -0,0 +1,263 @@
...
+
+    To achieve that, we define 4 keyframes. Between 0 and 10% of viewport,
+    video is fullsize, between 10% and 50%, it starts scaling down until it
+    settles at half its size when in the middle of the viewport and stays the
+    same size all the way to the last keyframe.
+
+    **Parallax background image**:
+    You may have noticed that the background image of this ad has a subtle parallax
```
Comment (sebastianbenz) [inline src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html]: s/You may have noticed that t/T/

File: src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html [hunk 8]
```diff
@@ -0,0 +1,263 @@
...
+    `amp-animation` to advance the animation timeline based on scrolling rather than time.
+
+    Normally animations have a duration defined, scrollbound animations also have
+    a duration but in pixels rather than milliseconds. Currently duration of
+    a scrollbound animation is approximately one viewport height, meaning all keyframes
+    of the animation will be hit between the time ad is fully in viewport until
+    user scrolls it partially out of the viewport.
+
+    **Scale down video to the corner when scrolling**:
+    As you have noticed, the video covers the whole ad initially and
```
Comment (sebastianbenz) [inline src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html]: * s/ As you have noticed, t/T/ 
* nit: `after the page is scrolled` -> `when the page scrolls` ?

File: src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html [hunk 9]
```diff
@@ -0,0 +1,263 @@
...
+        <amp-img class="poster-image" layout="fill" src="<%host%>/img/car-video-poster.png"></amp-img>
+      </div>
+
+      <amp-video id="video" controls layout="fill" src="<%host%>/video/car-video-360p.mp4"></amp-video>
+    </div>
+    <!-- ## Content Layer -->
+    <!--
+    Create the content layer that hosts the background image and information about the ad.
+
+    As seen above, we use `amp-animation` to simulate a parallax effect for
```
Comment (sebastianbenz) [inline src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html]: suggestion: "The `amp-animation` simulates a parallax..."

File: src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html [hunk 10]
```diff
@@ -0,0 +1,263 @@
...
+    `backgroundImg` when user scrolls.
```
Comment (sebastianbenz) [inline src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html]: s/user/the user ?

File: src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html [hunk 11]
```diff
@@ -0,0 +1,263 @@
...
+          ]
+        }
+      ]
+    }
+    </script>
+  </amp-animation>
+
+  <!-- ## Ad Container -->
+  <!--
+  Top level container: It fills the parent viewport that hosts the ad.
```
Comment (sebastianbenz) [inline src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html]: suggestion: "First, we declare a top level container which fills the parent viewport in order to host the ad."

File: src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html [hunk 12]
```diff
@@ -0,0 +1,263 @@
...
+  -->
+  <div class="ad-container">
+    <!-- ## Video Layer -->
+    <!--
+    Create a click-to-play overlay and a video element.
```
Comment (sebastianbenz) [inline src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html]: suggestion: "Inside the ad-container we create..."

File: src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html [hunk 13]
```diff
@@ -0,0 +1,263 @@
...
+
+    As seen above, we use `amp-animation` to scale down the `videoContainer` when user
```
Comment (sebastianbenz) [inline src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html]: s/As seen above, we use `amp-animation` to scale/The `amp-animation` scales down the ...

File: src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html [hunk 14]
```diff
@@ -0,0 +1,263 @@
...
+    scrolls.
+    -->
+    <div id="videoContainer" class="video-container fill">
+      <div id="clickToPlayOverlay" class="fill">
+        <div class="play-icon" role="button" tabindex="0" on="tap:clickToPlayOverlay.hide, video.play"></div>
+
+        <amp-img class="poster-image" layout="fill" src="<%host%>/img/car-video-poster.png"></amp-img>
```
Comment (sebastianbenz) [inline src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html]: please add image attribution

File: src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html [hunk 15]
```diff
@@ -0,0 +1,263 @@
...
+      </div>
+
+      <amp-video id="video" controls layout="fill" src="<%host%>/video/car-video-360p.mp4"></amp-video>
```
Comment (sebastianbenz) [inline src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html]: please add video attribution

File: src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html (hunk 14 above)
Comment (aghassemi) [inline src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html]: We have purchased and licensed these. Is there still a need for attribution?

**Comment by aghassemi** [comment]
All done. PTAL

File: src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html [hunk 16]
```diff
@@ -27,9 +26,19 @@
...
+    to create an ad that animates based on scrolling.
+
+
+    ![Demo of this example](https://cloud.githubusercontent.com/assets/2099009/26071260/4cfddfa2-395c-11e7-86d0-bb22fe0581eb.gif)
+  -->
+
   <!-- ## Styling -->
   <!--
-  This is a relatively advanced example that requires some styling to make it
+  This is a advanced example that requires some styling to make it
```
Comment (sebastianbenz) [inline src/amp-ads/40_Experimental_Ads/Scrollbound_Video_Ad.html]: s/a/an

File: gulpfile.js [hunk 17]
```diff
@@ -112,7 +112,7 @@ const config = {
     defaultHeight: 250,
     defaultLayout: "fixed"
   },
-  host: 'https://ampbyexample.com'
+  host: 'https://ampbyexample.firebaseapp.com'
```
Comment (sebastianbenz) [inline gulpfile.js]: can you undo this change?

Comment (aghassemi) [inline gulpfile.js]: :S sorry one.

**Comment by sebastianbenz** [comment]
Thanks!
